### PR TITLE
ocamlPackages.ssl: 0.5.10 -> 0.5.11

### PR DESCRIPTION
**Pull request by superherointj**
ocamlPackages.ssl: 0.5.10 -> 0.5.11

This PR is ready for review.

### CODE REVIEW

**Comment by ulrikstrid** [comment]
Can you add `cacert` as `checkInput`?

**Comment by superherointj** [comment]
> Can you add `cacert` as `checkInput`?

Just tested `cacert` but was not enough. Any other idea?

**Comment by dali99** [comment]
```
index 96b258a704d..ca31435c677 100644
--- a/pkgs/development/ocaml-modules/ssl/default.nix
+++ b/pkgs/development/ocaml-modules/ssl/default.nix
@@ -1,11 +1,10 @@
-{ alcotest
+{ lib
 , buildDunePackage
-, cacert
-, dune-configurator
 , fetchFromGitHub
-, lib
-, openssl
 , pkg-config
+, dune-configurator
+, openssl
+, alcotest
 }:
 
 buildDunePackage rec {
@@ -23,8 +22,12 @@ buildDunePackage rec {
   buildInputs = [ dune-configurator ];
   propagatedBuildInputs = [ openssl ];
 
-  doCheck = false; # SSL: Certificate error: couldn't open certificate file
-  checkInputs = [ alcotest cacert ];
+  checkInputs = [ alcotest ];
+  
+  preCheck = ''
+    mkdir -p _build/default/tests/
+    cp tests/digicert_certificate.pem _build/default/tests/
+  '';
 
   meta = {
     homepage = "http://savonet.rastageeks.org/";
```

It's looking for the test certificate in `_build/default/tests/` but it's not copied there when the test is compiled

**Comment by superherointj** [comment]
This PR is ready for review.

File: pkgs/development/ocaml-modules/ssl/default.nix [hunk 1]
```diff
@@ -1,30 +1,38 @@
...
+    sha256 = "sha256-uFr+XSKDGMHaM2o5DODYmt7+LkhnDzzlVX//CtAXBm4=";
   };
 
-  useDune2 = true;
   nativeBuildInputs = [ pkg-config ];
   buildInputs = [ dune-configurator ];
-  propagatedBuildInputs = [openssl];
+  propagatedBuildInputs = [ openssl ];
+
+  doCheck = true;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/ssl/default.nix:25]:
```suggestion
  doCheck = lib.versionAtLeast ocaml.version "4.05";
```

Comment (superherointj) [reply pkgs/development/ocaml-modules/ssl/default.nix:25]: I want to understand the process. Why `"4.06"`? Where does it come from? How did you figure out that?

Comment (vbgl) [reply pkgs/development/ocaml-modules/ssl/default.nix]: Don’t forget to take `ocaml` as argument.

Comment (vbgl) [reply pkgs/development/ocaml-modules/ssl/default.nix]: > Did you build all versions?

Yes.

> Is there a good way to build all versions?

None that I am aware of.

> I mostly use `nixpkgs-review pr` and test latest.

If you had run `nixpkgs-review` this time, you would have spotted that `opa` was gone. Investigating this lead to the suggested fix.

**Review by vbgl — APPROVED** [review]

**Pull request merged**